pkg/policy/api/rule_validation.go

```diff
@@ -37,19 +37,32 @@ func (r *Rule) Sanitize() error {
 		// Default posture is per-direction (ingress or egress),
 		// if there is a peer selector for that direction, the
 		// default is deny, else allow.
+		var hasEgress, hasIngress bool
 		if r.EnableDefaultDeny.Egress == nil {
-			x := len(r.Egress) > 0 || len(r.EgressDeny) > 0
-			r.EnableDefaultDeny.Egress = &x
+			hasEgress = len(r.Egress) > 0 || len(r.EgressDeny) > 0
+		} else {
+			hasEgress = *r.EnableDefaultDeny.Egress
 		}
+
 		if r.EnableDefaultDeny.Ingress == nil {
-			x := len(r.Ingress) > 0 || len(r.IngressDeny) > 0
-			r.EnableDefaultDeny.Ingress = &x
+			hasIngress = len(r.Ingress) > 0 || len(r.IngressDeny) > 0
+			if r.EnableDefaultDeny.Egress == nil && !hasEgress && !hasIngress {
```

Comment:
I think the nil check is redundant since `hasEgress` should be appropriately set if it is nil.
```suggestion
			if !hasEgress && !hasIngress {
```